Mathlib/FieldTheory/GalConj.lean

```diff
@@ -0,0 +1,323 @@
+/-
+Copyright (c) 2022 Yuyang Zhao. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yuyang Zhao
+-/
+import Mathlib.FieldTheory.IsAlgClosed.AlgebraicClosure
+import Mathlib.FieldTheory.NormalClosure
+
+/-!
+TODO
+-/
+
+variable (F E E' : Type*) [Field F] [Field E] [Algebra F E]
+  [Field E'] [Algebra F E'] [Algebra E E'] [IsScalarTower F E E']
+
+noncomputable abbrev NormalClosure : Type _ := normalClosure F E (AlgebraicClosure E)
+
+noncomputable irreducible_def NormalClosure.algebraAux [n : Normal F E'] :
+    NormalClosure F E →+* E' :=
+  let alglift : E' →ₐ[F] AlgebraicClosure E := IsAlgClosed.lift
+  haveI : Nonempty (E →ₐ[F] AlgebraicClosure E) := ⟨alglift.comp (IsScalarTower.toAlgHom F _ _)⟩
+  haveI : Algebra.IsAlgebraic F E := Algebra.IsAlgebraic.tower_bot _ _ E'
+  let aux : NormalClosure F E →ₐ[F] E' := IsNormalClosure.lift (K := E) fun x ↦ by
+    simpa [minpoly.algebraMap_eq, (algebraMap E E').injective] using n.splits (algebraMap _ _ x)
+  let comp : E →ₐ[F] E' := aux.comp (IsScalarTower.toAlgHom _ _ _)
+  ((AlgEquiv.ofBijective (comp.liftNormal E') (AlgHom.normal_bijective F _ _ _)).symm.toAlgHom.comp
+    aux).toRingHom
+
+noncomputable instance [Normal F E'] : Algebra (NormalClosure F E) E' :=
+  (NormalClosure.algebraAux F E E').toAlgebra
+
+instance [Normal F E'] : IsScalarTower E (NormalClosure F E) E' :=
+  .of_algebraMap_eq fun x ↦ by
+    change _ = NormalClosure.algebraAux _ _ _ _
+    simp [NormalClosure.algebraAux_def, AlgEquiv.eq_symm_apply]
+
+instance [Normal F E'] : IsScalarTower F (NormalClosure F E) E' :=
+  .of_algebraMap_eq fun x ↦ by
+    change _ = NormalClosure.algebraAux _ _ _ _
+    simp [NormalClosure.algebraAux_def, AlgEquiv.eq_symm_apply]
+
+def IsGalConj.setoid :=
+  (MulAction.orbitRel (NormalClosure F E ≃ₐ[F] NormalClosure F E) (NormalClosure F E)).comap
```

Comment:
This is quite a restrictive definition (only applies to NormalClosure). I recall it was defined using `minpoly` before; am I hallucinating? If #17783 is merged this file should probably switch to use it?

Reply:
I‘ve discussed with @FR-vdash-bot and we believe that the definition here and in #17783 is indeed mathematically equivalent, even in the case of transcendental extensions. (Although some more api's need to be built.) We agreed to use the `minpoly K x = minpoly K y` definition since it involves less.